azurelinuxagent/common/protocol/imds.py

```diff
@@ -49,10 +182,34 @@ def __init__(self,
         self.vmId = vmId
         self.vmSize = vmSize
 
+
     @property
     def image_info(self):
         return "{0}:{1}:{2}:{3}".format(self.publisher, self.offer, self.sku, self.version)
 
+    @property
+    def image_origin(self):
+        """
+        An integer value describing the origin of the image.
+
+          0 -> unknown
+          1 -> custom - user created image
+          2 -> endorsed - See https://docs.microsoft.com/en-us/azure/virtual-machines/linux/endorsed-distros
+          3 -> platform - non-endorsed image that is available in the Azure Marketplace.
+        """
+
+        try:
+            if self.publisher == "":
+                return IMDS_IMAGE_ORIGIN_CUSTOM
+
+            if ComputeInfo.__matcher.is_match(self.publisher, self.offer, self.sku, self.version):
+                return IMDS_IMAGE_ORIGIN_ENDORSED
+            else:
+                return IMDS_IMAGE_ORIGIN_PLATFORM
+
+        except Exception as e:
+            return IMDS_IMAGE_ORIGIN_UNKNOWN
```

Comment:
If we catch an exception here, we should almost certainly log it, rather than swallowing it silently. Consider emitting telemetry.

Reply:
I cannot easily emit a telemetry event, but I can log a warning.  The plan is create a telemetry event for all warnings and error in vNext+1.